docs/self-managed/deployment/helm/cloud-providers/kind.md

```diff
@@ -0,0 +1,466 @@
+
+While this guide uses kind, the same concepts apply to other local Kubernetes tools, like [K3s](https://k3s.io/), [minikube](https://minikube.sigs.k8s.io/), or [MicroK8s](https://microk8s.io/). The goal is to reduce the resources required by Camunda 8, so they can run on a personal machine.
+
+:::warning Local development only
+This setup is intended for **local development only** and shouldn't be used in production environments. For production deployments, follow our [cloud provider guides](/self-managed/deployment/helm/cloud-providers/index.md).
+:::
+
+If you encounter issues during deployment, refer to the [Troubleshooting](#troubleshooting) section.
+
+## Deployment modes
+
+You can choose from two deployment modes:
+
+| Mode          | Access                        | Requirements                    | Use case                              |
+| ------------- | ----------------------------- | ------------------------------- | ------------------------------------- |
+| **Domain**    | `https://camunda.example.com` | mkcert, hosts file modification | Full TLS setup, realistic environment |
+| **No-domain** | `localhost` via port-forward  | None                            | Quick setup, minimal configuration    |
+
+### How domain mode works
+
+In domain mode, you'll simulate a production-like environment locally with:
+
+- **Local DNS resolution**
+  - You'll configure your machine's `/etc/hosts` file to resolve `camunda.example.com` to `127.0.0.1`.
+  - Inside the cluster, you'll configure CoreDNS to rewrite DNS queries for this domain to the Ingress controller, allowing pods to communicate with each other using the same domain name.
+- **TLS certificates with mkcert**
+  - You'll use [mkcert](https://github.com/FiloSottile/mkcert) to generate locally-trusted certificates by installing a local Certificate Authority (CA) in your system's trust store. This allows your browser to trust the self-signed certificates without security warnings.
+  - You'll also mount the CA certificate in the pods that need to make HTTPS calls to other Camunda components.
+
+:::note Firefox compatibility
+mkcert requires [NSS](https://github.com/FiloSottile/mkcert#supported-root-stores) to work properly with Firefox. Without it, Firefox will display certificate errors. We recommend using **Chrome** or **Chromium-based browsers** for the best experience with domain mode.
+:::
+
+## Prerequisites
+
+Before you begin, ensure you've installed:
+
+- Docker ([Docker Desktop](https://www.docker.com/products/docker-desktop) or Docker Engine) with at least 4 CPU cores and 8 GB RAM available
+- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
+- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+- [Helm](https://helm.sh/docs/intro/install/)
+
+If you're deploying with **domain mode**, you'll also need:
+
+- Terminal access with administrator/sudo privileges for modifying `/etc/hosts`
```

Comment:
This is no longer domain mode specific, because we need to add camunda-keycloak to `/etc/hosts` in no-domain mode, too. I think we should make this one, big list:

Before you begin, you'll need:

- Terminal access with administrator/sudo privileges
- Docker ([Docker Desktop](https://www.docker.com/products/docker-desktop) or Docker Engine) with at least 4 CPU cores and 8 GB RAM available
- [kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation)
- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
- [Helm](https://helm.sh/docs/intro/install/)
- [mkcert](https://github.com/FiloSottile/mkcert#installation) (Domain mode only)